### PR TITLE
Use a Mosaic plugin to set lab status on component pages

### PR DESCRIPTION
**Pull request by DavieReid**
Hi @origami-z

### CODE REVIEW

**Review by JoanaMMoreira — APPROVED** [review]

File: site/mosaic.config.js [hunk 1]
```diff
@@ -23,6 +23,26 @@ const saltConfig = {
       modulePath: "@jpmorganchase/mosaic-plugins/BreadcrumbsPlugin",
       disabled: true,
     },
+    {
+      modulePath: require.resolve(
+        "./src/mosaic-plugins/LabsComponentPlugin.mjs"
+      ),
+      options: {
+        componentsList: [
+          "Badge",
```
Comment (JoanaMMoreira) [inline site/mosaic.config.js:32]: Badge seems to be missing the Lab icon for some reason

File: site/mosaic.config.js [hunk 2]
```diff
@@ -23,6 +23,26 @@ const saltConfig = {
...
+          "Combo Box",
+          "Dialog",
+          "Drawer",
+          "Dropdown",
+          "List",
+          "Pill",
+          "Stepped Tracker",
+          "Switch",
+        ],
+        statusLabel: "Lab component",
```
Comment (JoanaMMoreira) [inline site/mosaic.config.js:42]: I can't see the status label in the preview

**Review by joshwooding — COMMENTED** [review]
I'm not sure if I entirely understand the approach here. It seems strange that we've gone from the mdx being the source of truth for the component status to a list in a config?

File: site/mosaic.config.js (hunk 1 above)
Comment (DavieReid) [reply site/mosaic.config.js:32]: Should be fine now

File: site/mosaic.config.js (hunk 2 above)
Comment (DavieReid) [reply site/mosaic.config.js:32]: Where is it meant to appear?

**Comment by DavieReid** [comment]
> I'm not sure if I entirely understand the approach here. It seems strange that we've gone from the mdx being the source of truth for the component status to a list in a config?

Uses the page metadata now.  list of labs components has been removed

File: site/mosaic.config.js (hunk 2 above)
Comment (joshwooding) [reply site/mosaic.config.js:32]: Next to the title

File: site/src/mosaic-plugins/LabsComponentPlugin.mjs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  ) {
+    const isNonHiddenPage = createPageTest(ignorePages, pageExtensions);
+
+    for (const page of pages) {
+      if (!isNonHiddenPage(page.fullPath)) {
+        continue;
+      }
+
+      if (page.data?.package?.name === labPackageName) {
+        page.status = statusLabel;
```
Comment (joshwooding) [inline site/src/mosaic-plugins/LabsComponentPlugin.mjs:27]:
```suggestion
        page.data.status = statusLabel;
```
@DavieReid This should fix the badge

Comment (DavieReid) [reply site/src/mosaic-plugins/LabsComponentPlugin.mjs]: Thanks, just pushed a fix

**Review by joshwooding — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Review by mark-tate — APPROVED** [review]

**Pull request merged**